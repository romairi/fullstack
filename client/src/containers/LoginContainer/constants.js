export const EMAIL_FIELD = 'email';
export const EMAIL_FIELD_ERROR = 'email_error';
export const PASSWORD_FIELD = 'password';
export const PASSWORD_FIELD_ERROR = 'password_error';