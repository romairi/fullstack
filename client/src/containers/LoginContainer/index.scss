@import "../../scss/variables";

.base_container {
  .login_container {
    padding: 10px 30px;

    .title_login {
      text-align: center;
    }

    .server_error {
      color: $error_color;
      padding: 1rem 0;
      margin: 0 auto;
      text-align: center;
      width: 215px;
      word-wrap: break-word;
    }

    .form_login {
      display: flex;
      flex-direction: column;
      padding: 10px;

      .textField_email {
        display: inline-flex;
        align-items: flex-end;
        .icon_email {
          margin-right: 12px;
          font-size: 35px;
          color: $primary_color;
        }
      }

      .textField_password {
        display: inline-flex;
        align-items: flex-end;
        .icon_password {
          margin-right: 12px;
          font-size: 35px;
          color: $primary_color;
        }
      }
    }


    @media (max-width: $max-xs) {
      width: 95%;
    }
    @media (max-width: $max-sm) {
      width: 85%;
    }
  }
}


