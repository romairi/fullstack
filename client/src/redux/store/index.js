import {applyMiddleware, createStore} from 'redux'
import rootReducer from '../reducers';
import middleware from '../middleware';

const store = createStore(rootReducer, {}, applyMiddleware(...middleware));

export default store;