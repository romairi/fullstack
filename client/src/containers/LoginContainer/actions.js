import _ from 'lodash';
import {API_ACTION_TYPE, API_METHOD_POST} from "../../redux/middleware/ApiMiddleware/constants";
import {LOGIN_PATH} from "../../constants";

export function loginAction({data, onSuccess = _.noop, onError = _.noop}) {
    return {
        type: API_ACTION_TYPE,
        payload: {
            method: API_METHOD_POST,
            url: LOGIN_PATH,
            data,
            onSuccess,
            onError
        }
    };
}