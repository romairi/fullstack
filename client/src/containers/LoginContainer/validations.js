import Joi from "@hapi/joi";
import {EMAIL_FIELD, EMAIL_VALID_NAMES, PASSWORD_FIELD} from "./constants";

export const schemaLogin = Joi.object({
    [EMAIL_FIELD]: Joi.string()
        .email({ minDomainSegments: 2, tlds: { allow: EMAIL_VALID_NAMES } })
        .required(),
    [PASSWORD_FIELD]: Joi.string()
        .min(7)
        .required(),
});
