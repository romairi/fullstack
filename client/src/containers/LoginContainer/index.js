import React from 'react';
import './index.scss';
import _ from 'lodash';
import {useDispatch} from "react-redux";
import {Link} from 'react-router-dom';
import {replace} from "connected-react-router";
import Card from '@material-ui/core/Card';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import LockIcon from '@material-ui/icons/Lock';
import EmailIcon from '@material-ui/icons/Email';
import {EMAIL_FIELD, PASSWORD_FIELD} from "./constants";
import {BASE_ROUTE, SIGNUP_ROUTE} from '../../routes/constants';
import {schemaLogin} from './validations';
import useForm from "../../hooks/useForm";
import {loginAction} from './actions';
import {createSetUserAction} from "../../redux/reducers/UserReducer/actions";
import {GENERAL_ERROR_FIELD} from "../../hooks/constants";
import InputAdornment from "@material-ui/core/InputAdornment";


const Error = ({message}) => (
    message ?
        <p className='server_error'>
            {message}
        </p> : null
);


export default function LoginContainer({}) {

    const dispatch = useDispatch();

    const onSuccess = response => {
        dispatch(createSetUserAction(response.data));
        dispatch(replace(BASE_ROUTE));
    };

    const onSubmit = (values, handleServerError) => {
        dispatch(loginAction({data: values, onSuccess, onError: handleServerError}))
    };

    const {values, errors, handleChange, handleSubmit} = useForm(schemaLogin, onSubmit);
    return (
        <div className="base_container">
            <Card className="login_container">
                <h2 className="title_login"><strong>Login</strong></h2>
                <form className="form_login">
                    <TextField
                        className="text_field"
                        margin="dense"
                        error={!_.isEmpty(errors[EMAIL_FIELD])}
                        helperText={errors[EMAIL_FIELD]}
                        type="email"
                        id={EMAIL_FIELD}
                        label="Email"
                        value={values[EMAIL_FIELD] || ''}
                        onChange={handleChange}
                        InputProps={{
                            startAdornment: (
                                <InputAdornment position="start">
                                    <EmailIcon className="icon"/>
                                </InputAdornment>
                            ),
                        }}
                    />
                    <TextField
                        className="text_field"
                        margin="dense"
                        error={!_.isEmpty(errors[PASSWORD_FIELD])}
                        helperText={errors[PASSWORD_FIELD]}
                        type="password"
                        id={PASSWORD_FIELD}
                        label="Password"
                        value={values[PASSWORD_FIELD] || ''}
                        onChange={handleChange}
                        InputProps={{
                            startAdornment: (
                                <InputAdornment position="start">
                                    <LockIcon className="icon"/>
                                </InputAdornment>
                            ),
                        }}
                    />
                    <Error message={errors[GENERAL_ERROR_FIELD]}/>
                    <Button
                        className="base_container_btn"
                        variant="contained" color="primary"
                        type="submit"
                        onClick={handleSubmit}>
                        Sign In
                    </Button>
                </form>
                <div className="nav_another_page">
                    Don't have an account?
                    <Link to={SIGNUP_ROUTE}>
                        <strong>Sign up</strong>
                    </Link>
                </div>
            </Card>
        </div>
    );
}
