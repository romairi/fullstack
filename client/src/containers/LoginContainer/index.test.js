import React from 'react';
import renderer from 'react-test-renderer';
import LoginContainer from "./index";
import {Provider} from "react-redux";
import store from '../../redux/store';
import {BrowserRouter} from "react-router-dom";

// jest.mock('@material-ui/core/Card', () => () => <div className="Card"></div>);
jest.mock('@material-ui/core/Button', () => () => <div className="Button"/>);
jest.mock('@material-ui/core/TextField', () => () => <div className="TextField"/>);
jest.mock('@material-ui/icons/Lock', () => () => <div className="LockIcon"/>);
jest.mock('@material-ui/icons/Email', () => () => <div className="EmailIcon"/>);

describe('Login', () => {
    describe('render', () => {
        it('should render a component', () => {
            const component = renderer.create(
                <Provider store={store}>
                    <BrowserRouter>
                        <LoginContainer/>
                    </BrowserRouter>
                </Provider>
            );
            expect(component.toJSON()).toMatchSnapshot();
        });
    });
    describe('onSubmit', () => {

    });
});
