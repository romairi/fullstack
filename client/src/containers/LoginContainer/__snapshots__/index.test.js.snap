// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render should render a component 1`] = `
<div
  className="base_container"
>
  <div
    className="MuiPaper-root MuiCard-root login_container MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h2
      className="title_login"
    >
      Login
    </h2>
    <form
      className="form_login"
    >
      <div
        className="TextField"
      />
      <div
        className="TextField"
      />
      <div
        className="Button"
      />
    </form>
    <div
      className="nav_another_page"
    >
      Don't have an account?
      <a
        href="/signup"
        onClick={[Function]}
      >
        <strong>
          Sign up
        </strong>
      </a>
    </div>
  </div>
</div>
`;

exports[`Login render should render a component with snapshot 1`] = `
<div
  className="base_container"
>
  <div
    className="MuiPaper-root MuiCard-root login_container MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h2
      className="title_login"
    >
      Login
    </h2>
    <form
      className="form_login"
    >
      <div
        className="TextField"
      />
      <div
        className="TextField"
      />
      <div
        className="Button"
      />
    </form>
    <div
      className="nav_another_page"
    >
      Don't have an account?
      <a
        href="/signup"
        onClick={[Function]}
      >
        <strong>
          Sign up
        </strong>
      </a>
    </div>
  </div>
</div>
`;
